### PR TITLE
[camera_web] Add camera errors handling

### PR DESCRIPTION
**Pull request by bselwe**
Adds camera errors handling to the web implementation of the camera platform interface.

- Added media camera error codes to `CameraErrorCode` ([media error](https://developer.mozilla.org/en-US/docs/Web/API/MediaError/code)).
- Added `CameraWebException` thrown when the camera reports an initialization, configuration or video streaming error, or enters into an unexpected state.
- Moved `getMediaStreamForOptions` to `CameraSettings` - returns a media stream for the camera constrained by the given options.
- Added camera errors handling in the camera plugin.
  - Methods will now throw a `PlatformException` rather than a `DomException`/`CameraException` for consistency with the [MethodChannelCamera](https://github.com/flutter/plugins/blob/master/packages/camera/camera_platform_interface/lib/src/method_channel/method_channel_camera.dart) implementation. The only exemptions are `availableCameras` and `createCamera` that may throw a `CameraException`.
  - Each `CameraWebException` is caught and emitted on the `onCameraError` stream.
  - Added camera video element's [onError](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/onerror) and [onAbort](https://developer.mozilla.org/en-US/docs/Web/API/GlobalEventHandlers/onabort) events to the `onCameraError` stream.

Part of https://github.com/flutter/flutter/issues/45297.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Note that unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I [updated pubspec.yaml](https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates) with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code

### CODE REVIEW

**Comment by bselwe** [comment]
@ditman @felangel @orestesgaolin

File: packages/camera/camera_web/lib/src/camera_settings.dart [hunk 1]
```diff
@@ -8,8 +8,10 @@ import 'dart:ui';
 import 'package:camera_platform_interface/camera_platform_interface.dart';
 import 'package:camera_web/src/types/types.dart';
 import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
 
-/// A utility to fetch and map camera settings.
+/// A utility to fetch, map camera settings and
+/// obtain the camera stream.
 class CameraSettings {
```
Comment (ditman) [inline packages/camera/camera_web/lib/src/camera_settings.dart:15]: Is it time to find a better name for `CameraSettings`? Is it maybe doing too much? Slightly confusing naming with `CameraOptions`?

Maybe the parts regarding the management of the Camera Stream should be moved to a new class? Is this class more of a `CameraStreamsHelper`/`Handler`?

**Review by ditman — APPROVED** [review]
Let's keep the momentum going. Since `CameraSettings` is an internal implementation detail, it can be renamed/split in a further PR.

File: packages/camera/camera_web/lib/src/camera_settings.dart (hunk 1 above)
Comment (bselwe) [reply packages/camera/camera_web/lib/src/camera_settings.dart:15]: Good point! It is currently responsible for getting the camera stream based on `CameraOptions`, getting the facing mode setting based on the camera video track and mapping camera video track settings. How about renaming it to a `CameraService` (this way we will no longer mix settings with options)? 

I'm also okay with extracting the camera stream logic to a separate class (`CameraStreams.getStreamForOptions` maybe?) if you prefer it that way.

Comment (ditman) [reply packages/camera/camera_web/lib/src/camera_settings.dart:15]: `CameraService` sounds good to me. For the conversion functions, what I've been doing is to have a fat "convert.dart" with a bunch of functions there that translate from one type to my own (and vice-versa). See [convert.dart in google maps](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart). I normally prefer those functions to be stateless, so they can be just utility functions (they don't need to belong to any class or namespace).

**Pull request merged**